regression-tests.recursor-dnssec/test_EDNSBufferSize.py

```diff
@@ -0,0 +1,267 @@
+import dns
+import os
+import socket
+import struct
+import threading
+import time
+
+from recursortests import RecursorTest
+from twisted.internet.protocol import DatagramProtocol
+from twisted.internet import reactor
+
+ednsBufferReactorRunning = False
+
+class EDNSBufferTest(RecursorTest):
+    """
+    The tests derived from this one test several truncation related issues.
+    As an overview, this is what can be tested:
+
+        udp-truncation-threshold      edns-outgoing-bufsize
+                  |                           |
+    +------+      v      +----------+         v             +------------+
+    | stub | <=========> | recursor | <===================> | responders |
+    +------+             +----------+                       +------------+
+                 ^
+                 |
+      client bufsize (stub => recursor)
+     bufsize to client (recursor => stub)
+
+    The subclasses will test the following scenarios:
+
+    test | udp-trunc | edns-outgoing | client bufsize | response size   | result to client | bufsize to client |
+    -----+-----------+---------------+----------------+-----------------+------------------+-------------------+
+     01  | 1680      | 1680          | 4096           | 1680 (inc EDNS) | 1680 (inc EDNS)  | 1680              |
+     02  | 1680      | 1680          | 1679           | 1680 (inc EDNS) | TC (+EDNS)       | 1680              |
+     03  | 1680      | 1681          | 4096           | 1680 (inc EDNS) | TC (+EDNS)       | 1680              |
```

Comment:
Shouldn't the response size be 1681 for the third test?

Reply:
it is in my hand-written table and in the test as well.. fixing the table